### PR TITLE
Add `flip` tensor operator

### LINKED ISSUE #1099: Tensor flip along dimension
Thanks for the amazing library! :crab: 

### Feature description

I would like to flip a tensor along a given dimension. This would be like pytorch's [flip](https://pytorch.org/docs/stable/generated/torch.flip.html) function.

Basically it just creates a "mirror" image of the tensor:

> Reverse the order of an n-D tensor along given axis in dims.
>
> Parameters
> 
> input ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – the input tensor.
>
> dims (a list or [tuple](https://docs.python.org/3/library/stdtypes.html#tuple)) – axis to flip on

### Feature motivation

I want to create a reverse diagonal, but I'm sure there are many more usecases, as this seems like a very basic operation.

Forgive me if there's a way to do this that I missed.

### PR DESCRIPTION
**Pull request by carrotflakes**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

1. Fixes #1099
2. https://github.com/tracel-ai/burn/issues/1489 is improvement to add Candle native impl

### Changes

Added `flip` operation for tensors like the [pytorch one](https://pytorch.org/docs/stable/generated/torch.flip.html). It has also been implemented for all backends.

The implementation for the Candle backend may be slow, but it can be replaced with more suitable methods, such as flip or step indexing, once these are implemented on Candle.

### Testing

Added unit tests.

### CODE REVIEW

**Review by nathanielsimard — APPROVED** [review]
LGTM, seems pretty well done. Requesting @louisfd to double check before merging, but great job! 🎉

**Review by antimora — APPROVED** [review]
LGTM

**Comment by antimora** [comment]
Reviewed it myself and merged it so we can unblock new tensor op PRs.

**Pull request merged**
